### ISSUE TITLE
Extended newick support

### ISSUE THREAD

**Issue by jeromekelleher** [glottobank/python-newick#44]
I'd like to process trees output by StarBEAST, but it uses nested annotations that aren't supported. An example 2 leaf tree:
```python

import newick

tree = "(1[&dmv={1},dmv1=0.260,dmv1_95%_hpd={0.003,0.625},dmv1_median=0.216,dmv1_range={0.001,1.336},height=1.310e-15,height_95%_hpd={0.0,3.552e-15},height_median=0.0,height_range={0.0,7.105e-15},length=2.188,length_95%_hpd={1.725,2.634},length_median=2.182,length_range={1.307,3.236}]:1.14538397925438,2[&dmv={1},dmv1=0.260,dmv1_95%_hpd={0.003,0.625},dmv1_median=0.216,dmv1_range={0.001,1.336},height=1.310e-15,height_95%_hpd={0.0,3.552e-15},height_median=0.0,height_range={0.0,7.105e-15},length=2.188,length_95%_hpd={1.725,2.634},length_median=2.182,length_range={1.307,3.236}]:1.14538397925438)[&dmv={1},dmv1=0.260,dmv1_95%_hpd={0.003,0.625},dmv1_median=0.216,dmv1_range={0.001,1.336},height=1.310e-15,height_95%_hpd={0.0,3.552e-15},height_median=0.0,height_range={0.0,7.105e-15},length=2.188,length_95%_hpd={1.725,2.634},length_median=2.182,length_range={1.307,3.236}]"

newick.loads(tree)
```
This fails with:

```
    newick.loads(tree)
  File "/home/jk/.local/lib/python3.8/site-packages/newick.py", line 369, in loads
    return [parse_node(ss.strip(), **kw) for ss in s.split(';') if ss.strip()]
  File "/home/jk/.local/lib/python3.8/site-packages/newick.py", line 369, in <listcomp>
    return [parse_node(ss.strip(), **kw) for ss in s.split(';') if ss.strip()]
  File "/home/jk/.local/lib/python3.8/site-packages/newick.py", line 471, in parse_node
    return Node.create(name=name, length=length, descendants=descendants, **kw)
  File "/home/jk/.local/lib/python3.8/site-packages/newick.py", line 76, in create
    node = cls(name=name, length=length, **kw)
  File "/home/jk/.local/lib/python3.8/site-packages/newick.py", line 42, in __init__
    raise ValueError(
ValueError: Node names or branch lengths must not contain ","
```

It should be possible to make the parser a bit more sophisticated using regex so that stuff nested in the comment brackets isn't split.

Would it be possible to add this feature, or is it outside the remit of the package? Thanks.

**Comment by xrotwang** [comment]
I think we discussed this here https://github.com/dlce-eva/python-newick/pull/40
I'm not against including support for this format, in particular if we have an actual use case right now. I'd assume you'd want to have access to the annotation from the Node instance - correct? Could you provide me with such a newick string for testing?

**Comment by xrotwang** [comment]
You can at least get the topology:
```python
>>> newick.loads(tree, strip_comments=True)[0].newick
'(1:1.14538397925438,2:1.14538397925438)'
```

**Comment by xrotwang** [comment]
Re-reading our previous attempts, I'd now lean towards a two-step procedure:
1. Parse the tree ignoring comments.
2. Parse a (node name, comment) `dict`, and associate the comments with the nodes post-hoc.

**Comment by xrotwang** [comment]
Maybe a regex like
```python
NODE_NAME_WITH_COMMENT = re.compile(r'(?P<name>[^(),]+)\[(?P<comment>[^]]*)]')
```
is all that's needed for this second pass.

**Comment by jeromekelleher** [comment]
Seems like a perfectly reasonable approach @xrotwang. It's great to be able to pull out the topology, but I do need access to the comments (just as a string is the right approach, I think; the user can parse it themselves).

> I'm not against including support for this format, in particular if we have an actual use case right now. I'd assume you'd want to have access to the annotation from the Node instance - correct? Could you provide me with such a newick string for testing?

The string above should do? I'm happy to provide a few more examples, if you think we can implement the functionality.

**Comment by xrotwang** [comment]
I think I put together something that works: https://github.com/dlce-eva/python-newick/pull/45
Could you give it a go? It still needs support for writing the comment to the newick format - so that we get round-tripping.

**Comment by xrotwang** [comment]
Ok, supporting this was simpler than expected. Will add a couple more tests and release (with a changelog :) ) tomorrow.

**Comment by xrotwang** [comment]
Released with [1.3.0](https://pypi.org/project/newick/1.3.0/). 
- Docs: https://github.com/dlce-eva/python-newick#supported-newick-dialects
- CHANGELOG: https://github.com/dlce-eva/python-newick/blob/master/CHANGELOG.md :)

**Comment by jeromekelleher** [comment]
AMAZING! Thanks so much @xrotwang, this is super helpful for me :heart:
